var/spack/repos/builtin/packages/libzmq/package.py

```diff
@@ -5,120 +5,131 @@
+    deprecated("4.0.6", sha256="28a2a9c9b77014c39087a498942449df18bb9885cdb63334833525a1d19f2894")
+    deprecated("4.0.5", sha256="3bc93c5f67370341428364ce007d448f4bb58a0eaabd0a60697d8086bc43342b")
+
+    build_system("autotools", default="autotools")
     variant("docs", default=False, description="Build documentation")
-
+    variant("drafts", default=False, description="Build and install draft classes and methods")
     variant(
         "libbsd",
         when="@4.3.3:",
         default=(sys.platform != "darwin"),
-        description="Use strlcpy from libbsd " + "(will use own implementation if false)",
+        description="Use strlcpy from libbsd (will use own implementation if false)",
+    )
+    variant(
+        "libsodium",
+        default=True,
+        description="Build with message encryption support via libsodium",
     )
-
     variant("libunwind", default=False, description="Build with libunwind support")
 
-    depends_on("libsodium", when="+libsodium")
-    depends_on("libsodium@:1.0.3", when="+libsodium@:4.1.2")
-
     depends_on("autoconf", type="build", when="@master")
     depends_on("automake", type="build", when="@master")
     depends_on("libtool", type="build", when="@master")
-    depends_on("pkgconfig", type="build")
     depends_on("docbook-xml", type="build", when="+docs")
     depends_on("docbook-xsl", type="build", when="+docs")
+    depends_on("pkgconfig", type="build")
 
     depends_on("libbsd", when="+libbsd")
-
+    depends_on("libsodium", when="+libsodium")
+    depends_on("libsodium@:1.0.3", when="@:4.1.2 +libsodium")
     depends_on("libunwind", when="+libunwind")
 
-    conflicts("%gcc@8:", when="@:4.2.2")
-    conflicts("%gcc@11:", when="@:4.3.2")
-
     # Fix aggressive compiler warning false positive
     patch(
         "https://github.com/zeromq/libzmq/commit/92b2c38a2c51a1942a380c7ee08147f7b1ca6845.patch?full_index=1",
         sha256="310b8aa57a8ea77b7ac74debb3bf928cbafdef5e7ca35beaac5d9c61c7edd239",
         when="@4.3.3:4.3.4",
     )
-
     # Fix build issues with gcc-12
     patch(
         "https://github.com/zeromq/libzmq/pull/4334.patch?full_index=1",
         sha256="edca864cba914481a5c97d2e975ba64ca1d2fbfc0044e9a78c48f1f7b2bedb6f",
         when="@4.3.4",
     )
-
     # Fix static assertion failure with gcc-13
     patch(
         "https://github.com/zeromq/libzmq/commit/438d5d88392baffa6c2c5e0737d9de19d6686f0d.patch?full_index=1",
         sha256="e15a8bfe8131f3e648fd79f3c1c931f99cd896b2733a7df1760f5b4354a0687c",
         when="@4.3.3:4.3.4",
     )
 
-    def url_for_version(self, version):
-        if version <= Version("4.1.4"):
-            url = "http://download.zeromq.org/zeromq-{0}.tar.gz"
-        else:
-            url = "https://github.com/zeromq/libzmq/releases/download/v{0}/zeromq-{0}.tar.gz"
-        return url.format(version)
 
+class AutotoolsBuilder(autotools.AutotoolsBuilder):
```

Comment:
what's the point if there's just one build system?

Reply:
Anyways, removed it.